components/brave_rewards/resources/page/reducers/rewards_reducer.ts

```diff
@@ -191,11 +193,13 @@ const rewardsReducer: Reducer<Rewards.State | undefined> = (state: Rewards.State
     }
     case types.ON_CLEAR_ALERT: {
       let ui = state.ui
-      if (ui[action.payload.property] === undefined) {
+      const property: keyof typeof ui = action.payload.property
+      if (ui[property] === undefined) {
         break
       }
 
-      ui[action.payload.property] = null
+      // @ts-expect-error
+      ui[property] = null
```

Comment:
Can you add more to the comments where we've added expect-error about why we have those? Like a TODO for how to avoid it in the future.

Reply:
I've mostly updated the types where the removal of suppressImplicitAnyIndexErrors was causing errors but there are a few places I had to add @ts-expect-error to keep the existing behavior (which is the recommended solution https://www.typescriptlang.org/tsconfig/#suppressImplicitAnyIndexErrors)

Reply:
(sorry, must've added it to the PR description as you wrote this)

Reply:
I've used it where the fix would be quite complicated - here for example:

```ts

// we could do
ui[property as keyof typeof ui] = null


// but that property of `ui` might not be nullable (there are some non-nullable properties)
// so this would require a significant refactor of how the method works
```

Reply:
@petemill I've updated this (and others)